docs/spl-docs/time-modifiers.md

```diff
@@ -0,0 +1,177 @@
+
+When a snap to time unit isn't defined, the system defaults to using seconds. If an invalid snap to time unit is provided, it may result in a syntax error or unexpected search results.
+
+The snap to option becomes very useful in a range of situations. For example, if you want to search for events in the previous month, specify `earliest=-mon@mon latest=@mon`. This sets the search to start at the beginning of the last month and end at the beginning of this month.
+
+### Difference between relative time and relative snap to time
+
+Imagine it's May 15th, and you initiate a search at 10:30 AM.
+
+- If you use earliest=-3h, the search covers the past three hours, beginning at 7:30 AM on May 15th.
+- If you use earliest=-3h@h, the search adjusts to three hours ago but snaps to the start of the hour. This means the search includes events from 7:00 AM on May 15th.
+
+### Special time units
+
+The following abbreviations are reserved for special cases of time units and snap time offsets.
+
+
+
+| Time Unit | Description |
+|-----------|-------------|
+| `earliest=1` | To search events from the start of UNIX epoch time, use `earliest=1`. UNIX epoch time 1 is UTC January 1, 1970, at 12:00:01 AM. |
+| `earliest=0` | Including `earliest=0` in the search string indicates that time is not considered in the search. |
+| `earliest=1` and `latest=now` or `latest=<a_large_number>` | The search spans all time. However, specifying `latest=now` (the default) excludes future events, while `latest=<a_large_number>` includes future events, which are events with timestamps beyond the current time, `now()`. |
+| `earliest=now` or `latest=now` | Specifies that the search starts or ends at the current time. Any search including `earliest=<relative time offset>` should also include `latest=now`. For example, `earliest=-30s latest=now`. |
+| `@q`, `@qtr`, or `@quarter` | Specify a snap to the beginning of the most recent quarter: Jan 1, Apr 1, July 1, or Oct 1. |
+| `w0`, `w1`, `w2`, `w3`, `w4`, `w5`, `w6`, and `w7` | Specify "snap to" days of the week, with `w0` representing Sunday, `w1` for Monday, and so on. Snapping to a week using `@w` or `@week` defaults to snapping to Sunday, equivalent to `@w0`. Sunday can be referred to as either `w0` or `w7`. |
+
+### Specify earliest relative time offset and latest time in ad hoc searches
+
+Ad hoc searches that use the earliest time modifier with a relative time offset should also include `latest=now` in order to avoid time range inaccuracies. For instance, to capture all events from the last 15 seconds starting at 01:00:15, the following search returns all events that occur between the time of 01:00:00 and 01:00:15, as expected.
+
+```spl
+index=main earliest=-15s latest=now
+```
+
+Running the same search without including `latest=now` might produce unpredictable results.
```

Comment:
> Running the same search without including `latest=now` might produce unpredictable results.

Is this true? What could go wrong?

Reply:
I verified this on siglens we are automatically setting latest to `now` for all cases including ad-hoc searches. Since `latest=now` is set by default for all cases, I removed the section `Specify earliest relative time offset and latest time in ad hoc searches`.